semantic_conventions/trace/database.yaml

```
@@ -402,6 +402,94 @@ groups:
           value MUST NOT be set.
         examples: ['public.users', 'customers']
 
+  - id: db.cosmosdb
+    type: span
+    extends: db
+    prefix: db.cosmosdb
+    brief: >
+      Call-level attributes for CosmosDb.
+    attributes:
+      - id: client_id
+        type: string
+        brief: Unique cosmos client instance id.
+        examples: '3ba4827d-4422-483f-b59f-85b74211c11d'
+      - id: operation_type
+        type:
+          allow_custom_values: false
+          members:
+            - id: Invalid
+              value: 'Invalid'
+            - id: Create
+              value: 'Create'
+            - id: Patch
+              value: 'Patch'
+            - id: Read
+              value: 'Read'
+            - id: ReadFeed
+              value: 'ReadFeed'
+            - id: Delete
+              value: 'Delete'
+            - id: Replace
+              value: 'Replace'
+            - id: Execute
+              value: 'Execute'
+            - id: Query
+              value: 'Query'
+            - id: Head
+              value: 'Head'
+            - id: HeadFeed
+              value: 'HeadFeed'
+            - id: Upsert
+              value: 'Upsert'
+            - id: Batch
+              value: 'Batch'
+            - id: QueryPlan
+              value: 'QueryPlan'
+            - id: ExecuteJavaScript
+              value: 'ExecuteJavaScript'
+        brief: Cosmsodb Operation Type.
+        requirement_level: required
+      - ref: user_agent.cosmosdb
+      - id: connection_mode
+        type:
+          allow_custom_values: false
+          members:
+            - id: gateway
+              value: 'gateway'
+              brief: Gateway (HTTP) connections mode
+            - id: direct
+              value: 'direct'
+              brief: Direct connection.
+        brief: Cosmos client connection mode.
+        requirement_level: required
```

```suggestion
        requirement_level: 
          conditionally_required: if not `direct` (or pick gw as default)
```

having default value can save a bit for users on their telemetry bill

---

Just Curious, how will it save their telemetry bills?

---

it's common to charge for ingestion/storage depending on telemetry volume (which might not be linear or straightforward), but  usually, it costs more to get more telemetry, thus having a default value for something slightly decreases the volume of telemetry.